core/src/main/java/org/wso2/testgrid/core/ScenarioExecutor.java

```diff
@@ -73,6 +73,11 @@ public void runScenario(TestScenario testScenario, Deployment deployment, TestPl
             String testLocation = Paths.get(homeDir, testScenario.getName()).toAbsolutePath().toString();
             List<Test> tests = getTests(testScenario, testLocation);
 
+            // If the scenario directory doesn't have any JMeter scripts throw an Error
+            if (tests.isEmpty()) {
+                throw new ScenarioExecutorException("Scenario directory doesn't exist " + testScenario);
```

Comment:
We are throwing this same exception in `getTests` method. IMO `getTests` method should throw the exception and should be handled here (or later on accordingly).

Also, Does it matter if no tests are found? In the report, we can reflect this. IMO I do not this we need to do this validation WDYT?

Reply:
IMO we need to change the state of the scenario execution if there are no tests, coz scenarios without any tests don't make any sense. Rather than giving an error ideal way is to give a warning. Since there is no such state will remove this validation. :)

Reply:
+1 to log a warning here. We should highlight empty testcases in the dashboard as well.

Reply:
This is fixed.